### PR TITLE
[#IR-325] As a dev, Added support to use the JIRA time field in place of the TS.

### PR DESCRIPTION
**Pull request by chahat-arora**

### CODE REVIEW

File: src/app/shared/task-provider/task-provider.component.ts [hunk 1]
```diff
@@ -44,6 +45,7 @@ export class TaskProviderComponent implements OnInit {
     }
 
     onProviderChange(selectedProvider) {
+        this.getTimeProviders.emit();
```
Comment (himanshu-garg-165) [inline src/app/shared/task-provider/task-provider.component.ts]: We are not emitting any value here, but the type declaration mentions a string value.

File: src/app/retrospective/retrospective-create/retrospective-create.component.ts [hunk 2]
```diff
@@ -96,6 +101,31 @@ export class RetrospectiveCreateComponent implements OnInit, OnDestroy {
...
+
+        if (!team || !selectedTaskProvider) {
+            return;
+        }
+        this.disableTimeProviderField = false;
+        this.retrospectiveService.getTimeProvidersList(selectedTaskProvider, team)
+        .subscribe(
+            (data) => {
+                this.timeProvidersList = data.body.timeProviderList;
+                this.selectedTimeField = this.timeProvidersList[0];
```
Comment (himanshu-garg-165) [inline src/app/retrospective/retrospective-create/retrospective-create.component.ts]: Shouldn't we set the team level time provider as the pre-selected value?

File: src/app/shared/task-provider/task-provider.component.ts [hunk 3]
```diff
@@ -11,6 +11,7 @@ export class TaskProviderComponent implements OnInit {
 
     @Input() taskProviderOptions: any = [];
     @Output() initializedTaskProvider = new EventEmitter<FormGroup>();
+    @Output() getTimeProviders = new EventEmitter<string>();
```
Comment (himanshu-garg-165) [inline src/app/shared/task-provider/task-provider.component.ts]: `onProviderChanged` looks better here, we should not use "time provider" keyword anywhere in the task provider component.
```suggestion
    @Output() onProviderChanged = new EventEmitter<string>();
```

File: src/app/retrospective/retrospective-create/retrospective-create.component.ts (hunk 2 above)
Comment (chahat-arora) [reply src/app/retrospective/retrospective-create/retrospective-create.component.ts]: discussed.

File: src/app/shared/task-provider/task-provider.component.ts (hunk 1 above)
Comment (chahat-arora) [reply src/app/shared/task-provider/task-provider.component.ts]: done.

File: src/app/shared/task-provider/task-provider.component.ts (hunk 3 above)
Comment (chahat-arora) [reply src/app/shared/task-provider/task-provider.component.ts]: done.

File: src/app/retrospective/retrospective-create/retrospective-create.component.html [hunk 4]
```diff
@@ -58,13 +58,29 @@ <h2 mat-dialog-title>Create Team Retrospective</h2>
...
     </div>
-
+    <mat-toolbar class="small-toolbar">
+        Time Provider
+    </mat-toolbar>
+    <div class="table-box padded-container">
+        <div class="flex-container layout-row">
+            <div class="flex">
+                <mat-form-field>
+                        <mat-hint *ngIf="disableTimeProviderField" class="project-name-hint">Choose a Task Provider and Team first</mat-hint>
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/retrospective/retrospective-create/retrospective-create.component.html]: Renamed.

File: src/app/retrospective/retrospective-create/retrospective-create.component.ts [hunk 5]
```diff
@@ -29,6 +31,9 @@ export class RetrospectiveCreateComponent implements OnInit, OnDestroy {
     // Maintains the task providers count
     taskProvidersIndex = -1;
 
+    timeProvidersList: any = [];
+    disableTimeProviderField = true;
+    selectedTimeField = {};
```
[the remark this answers is not in the document]
Comment (chahat-arora) [reply src/app/retrospective/retrospective-create/retrospective-create.component.ts]: Done.

**Pull request merged**